lib/streamlit/config.py

```diff
@@ -711,12 +711,12 @@ def _server_address() -> str | None:
 _create_option(
     "server.enableCORS",
     description="""
-        Enables support for Cross-Origin Resource Sharing (CORS) protection, for
-        added security.
+        Enables support for Cross-Origin Resource Sharing (CORS) protection,
+        for added security.
 
-        Due to conflicts between CORS and XSRF, if `server.enableXsrfProtection` is
-        on and `server.enableCORS` is off at the same time, we will prioritize
-        `server.enableXsrfProtection`.
+        XSRF protection requires CORS protection. If XSRF protection is
+        enabled and CORS protection is disabled at the same time, Streamlit
+        will prioritize XSRF protection and use `server.enableCORS=True`.
```

Comment:
Actually, after looking at this again, I am not sure that this is true. 

the deleted paragraph above correctly describes the behavior of our code; however, in terms of the reasons why this is so, I was unable to find confirmation that CSRF protection only works if the Cross-Origin Resource Sharing is off.

We need to delve into history to understand why these configurations were written this way and, if possible, decouple them.
 
For now, I think it would be better to remove this paragraph because perhaps it does not correspond to reality in terms of how HTTP protocol works.